eldritch/ancient_ones.py

```diff
@@ -0,0 +1,169 @@
+import abc
+
+from eldritch import events, places, characters, monsters, values
+
+
+class AncientOne(metaclass=abc.ABCMeta):
+  # pylint: disable=unused-argument
+  def __init__(self):
+    self.doom = 0
+    self.combat_difficulty = 0
+    self.attributes = []
+
+  def get_interrupts(self, owner, state):
+    return []
+
+  def get_modifier(self, thing, attribute):
+    return 0
+
+  def get_override(self, other, attribute):
+    return None
+
+  def awaken(self, state):
+    pass
+
+  @abc.abstractmethod
+  def attack(self, state):
+    raise NotImplementedError
+
+
+class DummyAncient(AncientOne):
+  def attack(self, state):
+    pass
+
+
+class SquidFace(AncientOne):
+  def __init__(self):
+    super().__init__()
+    self.name = "SquidFace"
+    self.max_doom = 13
+    self.combat_difficulty = -6
+
+  def get_modifier(self, thing, attribute):
+    if isinstance(thing, monsters.Cultist):
+      return {"horror": -2, "horror_damage": 2}.get(attribute, 0)
+    if isinstance(thing, characters.Character):
+      return {"maximum_sanity": -1, "maximum_stamina": -1}
+    return 0
+
+  def attack(self, state):
+    self.doom = min(self.doom + 1, self.max_doom)
```

Comment:
The more I look at this, the more I think that the attack needs to be an event, and not just some code. For example, the redeemed cultist has an ability that needs to trigger when a doom token is added to the doom track.

Reply:
I put some todo's in the obvious places (here and in OpenGate) to make an AddDoom event. Leaving the doom modifier here for now.